### DISCUSSION TITLE
How does this application work?

### DISCUSSION THREAD

**Discussion by Silvergish** [couldbejake/spotify2mp3#46] [Q&A] [answered]
I'm trying to understand. Does this app take the Spotify link you input and then search YouTube for the song(s) and download them from there?

**Comment by couldbejake** [comment, marked as answer]
Hi of course.

The application uses a Spotify API to search for playlists either liked or saved, public or private.

The application then searches YouTube for the best matching song, then adds metadata, and song covers.

A folder is then created with output.

**Reply by RaptaG** [reply]
How is the best match determined? Also shouldn't it be using YT Music and YouTube as a fallback?

**Comment by Silvergish** [comment]
Thanks! Very cool. :)
